File: CPP/RemoveElement.cpp
class Solution {
public:
    int removeElement(int A[], int n, int elem) {
        // IMPORTANT: Please reset any member data you declared, as
        // the same Solution instance will be reused for each test case.
        int del_count = 0;
        for (int i = 0; i < n; ++i)
        {
        	if (A[i] == elem)
        	{
        		del_count++;
        	} else {
        		A[i - del_count] = A[i];
        	}
        }
        return n - del_count;
    }
};

File: JAVA/RemoveElement.java
public class Solution {
    public int removeElement(int[] A, int elem) {
        int size = A.length;
        int removeCount = 0;
        for (int i = 0; i < size - removeCount; i++) {
            if (A[i] == elem) {
                int temp = A[i];
                A[i] = A[size - removeCount - 1];
                A[size - removeCount - 1] = temp;
                removeCount++;
                i--;
            }
        }
        return size - removeCount;
    }
}
